src/JustEat.StatsD/PooledUdpTransport.cs

```diff
@@ -40,66 +99,58 @@ public void Send(string metric)
 
             var bytes = Encoding.UTF8.GetBytes(metric);
             var endpoint = _endpointSource.GetEndpoint();
+            var pool = GetPool(endpoint);
+            var socket = pool.Pool.PopOrCreate();
 
-            var socket = _socketPool.PopOrCreate();
             try
             {
-                socket.SendTo(bytes, endpoint);
+                socket.Send(bytes);
             }
             catch (Exception)
             {
                 socket.Dispose();
                 throw;
             }
 
-            _socketPool.Push(socket);
+            pool.Pool.Push(socket);
         }
-
-        /// <summary>
-        /// Finalizes an instance of the <see cref="PooledUdpTransport"/> class.
-        /// </summary>
-        ~PooledUdpTransport()
```

Comment:
The `IDisposable` implementation should be retained.

Reply:
It's a sealed class, so there would be no point to this dispose pattern.